### ISSUE TITLE
Arrows and segfault and concurrent live sessions

### ISSUE THREAD

**Issue by afortiorama** [ganga-devs/ganga#235]
I often use the arrows to go up and down the history of commands. Sometimes I click on the down arrow one time too many and ganga segfaults. I then have to wait few tenths of seconds to restart it. 

I tried with python and python shell doesn't crash for the extra clicks. This gives really the impression of fragility to be honest.

Also when one mistakenly starts more than one ganga session, it would be good to have a "steal" option or get an error message maybe proposing to close the other session and the current session to exit before it tries to start anything. Instead of this long series of messages for several minutes. After all ganga knows it in the first two lines there is another live session. 

```
*** Welcome to Ganga ***
Version: Ganga-6-1-2-hotfix1
Documentation and support: http://cern.ch/ganga
Type help() or help('index') for online help.

This is free software (GPL), and you are welcome to redistribute it
under certain conditions; type license() for details.


Ganga.Utility.Config               : INFO     reading config file /home/aforti/.gangarc
Ganga.Core.GangaRepository         : WARNING  1 other concurrent sessions:
 * vm7.tier2.hep.manchester.ac.uk.06.59_Saturday_27_February_2016.PID.7192.session
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs


In [1]:exit
Out[1]: 
<function exit at 0x7f89c40b7488>
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs

In [2]:
Do you really want to exit ([y]/n)? y
Job status update or output download still in progress (shutdown not completed after 6 seconds).
2 background thread(s) still running: ['GangaTasks', 'JobRegistry_Monitor'].
Do you want to force the exit (y/[n])? y
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GangaThread                  : WARNING  Shutdown forced. 2 background thread(s) still running: ['GangaTasks', 'JobRegistry_Monitor']
Ganga.GPIDev.Lib.Registry          : INFO     conf-afccaff7-968b-4bb1-a6c3-6944ae2864a9 not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : ERROR    Failed to get access to registry: jobs. Reason: RegistryLockError: Could not lock 'jobs' object #245! Object is locked by session 'vm7.tier2.hep.manchester.ac.uk.06.59_Saturday_27_February_2016.PID.7192.session.jobs.locks' 
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : ERROR    Failed to get access to registry: jobs. Reason: RegistryLockError: Could not lock 'jobs' object #246! Object is locked by session 'vm7.tier2.hep.manchester.ac.uk.06.59_Saturday_27_February_2016.PID.7192.session.jobs.locks' 
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Lib.Registry          : INFO     conf-52ebd195-4fbd-43af-a591-97a20085ba1c not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Lib.Registry          : INFO     conf-6960de48-0908-4c40-ba2d-179e9b285075 not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Lib.Registry          : INFO     conf-9acf20d6-6b91-474f-b06c-99157079816d not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Lib.Registry          : INFO     conf-4fe62ac9-2c34-4516-bdcc-fee7a45c9ade not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : ERROR    Failed to get access to registry: jobs. Reason: RegistryAccessError: Cannot get write access to a disconnected repository!
Exception in thread GANGA_Update_Thread_JobRegistry_Monitor:
Traceback (most recent call last):
  File "/home/aforti/dirac_ui/Linux_x86_64_glibc-2.12/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/cvmfs/ganga.cern.ch/Ganga/install/6.1.2-hotfix1/python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py", line 383, in run
    self.__monStep()
  File "/cvmfs/ganga.cern.ch/Ganga/install/6.1.2-hotfix1/python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py", line 432, in __monStep
    cbHookFunc(**cbHookEntry.argDict)
  File "/cvmfs/ganga.cern.ch/Ganga/install/6.1.2-hotfix1/python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py", line 773, in f
    activeBackends = activeBackendsFunc()
  File "/cvmfs/ganga.cern.ch/Ganga/install/6.1.2-hotfix1/python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py", line 697, in __defaultActiveBackendsFunc
    j._getWriteAccess()
  File "/cvmfs/ganga.cern.ch/Ganga/install/6.1.2-hotfix1/python/Ganga/GPIDev/Base/Objects.py", line 531, in _getWriteAccess
    raise x
RegistryAccessError: RegistryAccessError: Cannot get write access to a disconnected repository!

Ganga.Core.MonitoringComponent     : INFO     Stopping the monitoring component...
Ganga.GPIDev.Lib.Registry          : INFO     conf-afccaff7-968b-4bb1-a6c3-6944ae2864a9 not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Lib.Registry          : INFO     conf-a249792d-c511-40d4-9d5f-f6ff02b9026c not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Lib.Registry          : INFO     conf-a5d68f58-3201-41ad-84e7-0f4372c6b3df not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Lib.Registry          : INFO     conf-bf245d14-5aa6-4e30-8bd6-336249f7401b not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Lib.Registry          : INFO     conf-45a98be2-f1ba-4898-acd4-52def95fe77f not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Lib.Registry          : INFO     conf-a203736e-92ca-4fae-875b-d45fe952f693 not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Lib.Registry          : INFO     conf-3ced886a-3222-45da-8968-13426beb423b not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Lib.Registry          : INFO     conf-f0c18ae6-6e29-42b8-b05e-eb6edfad49f4 not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Lib.Registry          : INFO     conf-fd39ee6e-349d-44c5-ba5f-64fd657c5acc not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : ERROR    Failed to get access to registry: prep.metadata. Reason: RegistryLockError: Could not lock 'prep.metadata' object #0! Object is locked by session 'vm7.tier2.hep.manchester.ac.uk.06.59_Saturday_27_February_2016.PID.7192.session.prep.metadata.locks' 
Ganga.Runtime                      : ERROR    Failed to Shutdown prep Repository!!! please check for stale lock files
Ganga.Runtime                      : ERROR    Trying to shutdown cleanly regardless
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Lib.Registry          : INFO     conf-a42db22d-ce10-4f4e-8fed-16888b32f59b not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Lib.Registry          : INFO     conf-a008d656-4725-4ea7-ae45-33d440ccea85 not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Lib.Registry          : INFO     conf-52ebd195-4fbd-43af-a591-97a20085ba1c not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Lib.Registry          : INFO     conf-6960de48-0908-4c40-ba2d-179e9b285075 not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Lib.Registry          : INFO     conf-9acf20d6-6b91-474f-b06c-99157079816d not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Lib.Registry          : INFO     conf-4fe62ac9-2c34-4516-bdcc-fee7a45c9ade not found on disk. Removing entry from shareref table and unpreparing any associated Ganga objects.
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: jobs
Ganga.GPIDev.Base.Objects          : ERROR    Failed to get access to registry: jobs. Reason: RegistryAccessError: Cannot get write access to a disconnected repository!
Exception in thread GANGA_Update_Thread_JobRegistry_Monitor:
Traceback (most recent call last):
  File "/home/aforti/dirac_ui/Linux_x86_64_glibc-2.12/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/cvmfs/ganga.cern.ch/Ganga/install/6.1.2-hotfix1/python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py", line 383, in run
    self.__monStep()
  File "/cvmfs/ganga.cern.ch/Ganga/install/6.1.2-hotfix1/python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py", line 432, in __monStep
    cbHookFunc(**cbHookEntry.argDict)
  File "/cvmfs/ganga.cern.ch/Ganga/install/6.1.2-hotfix1/pythGanga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : INFO     Waiting on Write access to registry: prep.metadata
Ganga.GPIDev.Base.Objects          : ERROR    Failed to get access to registry: prep.metadata. Reason: RegistryLockError: Could not lock 'prep.metadata' object #0! Object is locked by session 'vm7.tier2.hep.manchester.ac.uk.06.59_Saturday_27_February_2016.PID.7192.session.prep.metadata.locks' 
Ganga.Runtime                      : ERROR    Failed to Shutdown Repository: prep !!! please check for stale lock files
Ganga.Runtime                      : ERROR    RegistryLockError: Could not lock 'prep.metadata' object #0! Object is locked by session 'vm7.tier2.hep.manchester.ac.uk.06.59_Saturday_27_February_2016.PID.7192.session.prep.metadata.locks' 
Ganga.Runtime                      : ERROR    Trying to Shutdown cleanly regardless
```

**Comment by rob-c** [comment]
Concurrent live sessions are not supported.
There are caveats around this but to first order concurrent sessions aren't supported.
You get a line at the top warning you in red that another session is running. Do you not see this or are you proposing we make some modifications to this?

This will likely not change in the near future as we make extensive use of an XML repository to store job information on disk. What you're seeing is a result of trying to access the same ```prep``` repository from multiple instances at the same time which isn't supported.

Here, the ```prep``` repository is locked by one Ganga session due to Ganga currently having no safe mechanism for releasing access to the various repositories on disk after actions have finished with it. (This is a long standing issue relating to many issues around both concurrent access and the prep repository being not particularly well understood).
When a second Ganga session attempts use this the same repository the access will fail which is what you're seeing above.

Or,
What you're seeing is that after Ganga has exited uncleanly the session lock files are persisting from one session to another.
This can be fixed by removing the session lockfiles under ```gangadir/repository```. This is automatically done by Ganga once the lockfile is reasonably stale, however Ganga has to run on systems with _very_ slow file access so the timeout on lockfiles is set to be something similar to 30sec, which means that launching Ganga immediately after it crashes causes issues similar to what you see above.


The history management within a 'live' Ganga session is managed solely within the IPython external package. (Currently IPython 1.2.1 for Ganga 6.1.16)
We're looking to upgrade this to IPython 4.x in the near future but at the moment is limited by our dependence on Python 2.6 for our target environments.
As of 6.1.2 I think you're still using a very old version of IPython (0.6.12 if memory serves) and I can't comment on what may or may not be broken in this for different systems.

**Comment by rob-c** [comment]
@afortiorama What sort of system are you seeing IPython crashes on?

**Comment by afortiorama** [comment]
SL6 I'm using ganga 6.1.2.

**Comment by afortiorama** [comment]
About the previous comment those sort of messages happen when I try to get out of the second ganga shell. I don't want concurrent shells. I'm just suggesting that ganga shouldn't even start if it detects a live version (not a dead one) or should have the ability to still it. As it is I often end up doing a kill <proc_id> rather than waiting for ganga to exit several minutes later.

**Comment by rob-c** [comment]
@afortiorama I think you bring up a good point and I'll open a separate issue to discuss this particular problem in more detail.

I'd like to keep this issue open for a while so that all everyone is aware of ```prep.metadata``` like issues that we do see from time to time.

I know it's a recurring theme but if you can try the 6.1.16 release addresses several concurrency problems. As well as thread safety problems within the same instance of Ganga which can lead to some of the problems you're seeing. (and other things such as Ganga appearing to lock up on shutdown).

**Comment by rob-c** [comment]
OK, we now have stopped Ganga from launching a monitoring thread if it determines that there are still some lock files on disk (zombie or not) so you won't be greeted with cannot lock object errors on launch until you try to access a locked object. (Then I don't think there is much we can do).
The startup/shutdown should have been improved a lot recently so you should find less reasons to have to force quit Ganga on exit after 6.1.21

**Comment by afortiorama** [comment]
I think this can be closed. It hasn't happened again with the new versions.

**Comment by egede** [comment]
@afortiorama Thanks for the feedback.
